### ISSUE TITLE
Installation issue: arborx

### ISSUE THREAD

**Issue by jilpi** [LLNL/spack#16411]
The latest commit concerning arborx' package ([598c233](https://github.com/spack/spack/commit/598c233f78d23329c0cc6c7bc86b7117dca7bacd)) doesn't work:
```console
$ spack install arborx
[...]
==> 916127: arborx: Executing phase: 'cmake'
==> Error: KeyError: 'No spec with name kokkos in arborx@0.8-beta2%gcc@9.3.0 [...]
/home/picard/spack/var/spack/repos/builtin/packages/arborx/package.py:39, in cmake_args:
         36        spec = self.spec
         37
         38        options = [
  >>     39            '-DCMAKE_PREFIX_PATH=%s' % spec['kokkos'].prefix,
         40            '-DARBORX_ENABLE_TESTS=OFF',
```

Can be fixed by changing `package.py` as follows:
```python
    options = [
      '-DCMAKE_PREFIX_PATH=%s' % spec['kokkos-legacy'].prefix,
```

I will try to submit a PR.
Thanks

**Comment by jilpi** [comment]
Here's the PR for this issue: https://github.com/spack/spack/pull/16413

**Comment by alalazo** [comment]
Fixed by #16413
